src/main/cpp/tiv.cpp

```diff
@@ -440,16 +439,24 @@ cimg_library::CImg<unsigned char> load_rgb_CImg(const char * const filename) {
 }
 
 int main(int argc, char* argv[]) {
+  int maxWidth = 80;
```

Comment:
shouldn't these be detected, or passed in somehow?  I mean, I'm okay with defaults, but just my first thought here.

Reply:
The actual dimensions are detected just a few lines below. 80x24 is just a fallback in case both those `ioctl` calls fail (Your current behavior is to go with whatever `ioctl` returned, even if it is 0x0 or some weird initialized state like -4x1). These variables are first declared long before they are used normally just because I didn't want to entwine [platform specific code with platform agnostic one too much](https://github.com/crouvpony47/tiv-win/blob/c035cba17ace73c2b98d6a01299d948522b84c01/src/main/cpp/tiv.cpp#L510).

Reply:
As for the "passed in somehow" part: my patch doesn't change this in any way, `-h` and `-w` parameters should work exactly as they did. It only fixes automatic size detection in case stdout is redirected to a file